lxd/instance/drivers/driver_qemu.go

```diff
@@ -5716,7 +5716,49 @@ func (d *qemu) CanMigrate() (bool, bool) {
 
 // DeviceEventHandler handles events occurring on the instance's devices.
 func (d *qemu) DeviceEventHandler(runConf *deviceConfig.RunConfig) error {
-	return fmt.Errorf("DeviceEventHandler Not implemented")
+	if !d.IsRunning() {
+		return nil
+	}
+
+	if runConf == nil || len(runConf.Uevents) == 0 {
+		return nil
+	}
+
+	// Uevents will contain 1 entry at most, therefore we don't need to iterate through it.
+	for _, event := range runConf.Uevents[0] {
```

Comment:
What about if it contains zero entries, LXD will crash?

Reply:
That's checked a few lines above.

Reply:
Ah so it is.